docs/extended_api/synchronization_library/pipeline.md

```diff
@@ -0,0 +1,104 @@
+| [producer_commit](./pipeline/producer_commit.md)   | commits operations previously issued by the current thread to the current _pipeline stage_                                                       |
+| [consumer_wait](./pipeline/consumer_wait.md)       | blocks the current thread until all operations committed to the current _pipeline stage_ complete                                                |
+| [consumer_wait_for](./pipeline/consumer_wait.md)   | blocks the current thread until all operations committed to the current _pipeline stage_ complete or after the specified timeout duration        |
+| [consumer_wait_until](./pipeline/consumer_wait.md) | blocks the current thread until all operations committed to the current _pipeline stage_ complete or until specified time point has been reached |
+| [consumer_release](./pipeline/consumer_release.md) | release the current _pipeline stage_                                                                                                             |
+| [quit](./pipeline/quit.md)                         | quits current thread's participation in the _pipeline_                                                                                           |
+
+## Notes
+
+A thread role cannot change during the lifetime of the pipeline object.
+
+## Example
+
+```c++
+#include <cuda/pipeline>
+#include <cooperative_groups.h>
+
+// Disables `pipeline_shared_state` initialization warning
+#pragma diag_suppress static_var_with_dynamic_init
+
+template <typename T>
+__device__ void compute(T * ptr);
+
+template <typename T>
+__global__ void example_kernel(T * global1, T * global2, size_t subset_count)
+{
+    extern __shared__ T s[];
+    auto group = cooperative_groups::this_thread_block();
+    T * shared[2] = { s, s + 2 * group.size() };
+
+    // Create a CUDA pipeline
+     constexpr unsigned stages_count = 2;
+    __shared__ cuda::pipeline_shared_state<cuda::thread_scope::thread_scope_block, stages_count> shared_state;
+    auto pipeline = cuda::make_pipeline(group, &shared_state);
+
+    // Prime the pipeline
+    pipeline.producer_acquire();
+    cuda::memcpy_async(group, shared[0],
+                       &global1[0], sizeof(T) * group.size(), pipeline);
+    cuda::memcpy_async(group, shared[0] + group.size(),
+                       &global2[0], sizeof(T) * group.size(), pipeline);
+    pipeline.producer_commit();
+
+    // Pipelined copy/compute
+    for (size_t subset = 1; subset < subset_count; ++subset) {
+        pipeline.producer_acquire();
+        cuda::memcpy_async(group, shared[subset % 2],
+                           &global1[subset * group.size()], sizeof(T) * group.size(), pipeline);
+        cuda::memcpy_async(group, shared[subset % 2] + group.size(),
+                           &global2[subset * group.size()], sizeof(T) * group.size(), pipeline);

```

Comment:
Reformat the calls to `memcpy_async` so that a line split does not occur between the destination and source arguments.